internal/api/attestationconfigapi/reporter.go

```diff
@@ -0,0 +1,197 @@
+		return fmt.Errorf("reporting version: %w", err)
+	}
+	if force {
+		if err := c.uploadAzureSEVSNPVersion(ctx, inputVersion, now); err != nil {
+			return fmt.Errorf("force uploading version: %w", err)
+		}
+		return nil
+	}
+	versionDates, err := c.listReportedVersions(ctx)
+	if err != nil {
+		return fmt.Errorf("list reported versions: %w", err)
+	}
+	if len(versionDates) < c.cacheWindowSize {
+		c.s3Client.Logger.Infof("Skipping version update since found only %d out of expected reported versions.", len(versionDates), c.cacheWindowSize)
+		return nil
+	}
+	minVersion, minDate, err := c.findMinVersion(ctx, versionDates)
+	if err != nil {
+		return fmt.Errorf("get minimal version: %w", err)
+	}
+	c.s3Client.Logger.Infof("Found minimal version: %+v with date: %s", *minVersion, minDate)
+	shouldUpdateAPI, err := isInputNewerThanOtherVersion(*minVersion, latestAPIVersion)
+	if err == nil && shouldUpdateAPI {
+		c.s3Client.Logger.Infof("Input version: %+v is newer than latest API version: %+v", *minVersion, latestAPIVersion)
+		t, err := time.Parse(VersionFormat, minDate)
+		if err != nil {
+			return fmt.Errorf("parsing date: %w", err)
+		}
+		if err := c.uploadAzureSEVSNPVersion(ctx, *minVersion, t); err != nil {
+			return fmt.Errorf("uploading version: %w", err)
+		}
+		c.s3Client.Logger.Infof("Successfully uploaded new Azure SEV-SNP version: %+v", *minVersion)
+		return nil
+	}
+	c.s3Client.Logger.Infof("Input version: %+v is not newer than latest API version: %+v", *minVersion, latestAPIVersion)
+	return nil
+}
+
+// SetCacheVersionSize sets a custom number of versions to be considered for the latest version.
+func (c *Client) SetCacheVersionSize(size int) {
+	c.cacheWindowSize = size
+}
+
+// reportAzureSEVSNPVersion uploads the latest observed version numbers of the Azure SEVSNP. This version is used to later report the latest version numbers to the API.
+func (c Client) reportAzureSEVSNPVersion(ctx context.Context, version AzureSEVSNPVersion, date time.Time) error {
+	dateStr := date.Format(VersionFormat) + ".json"
+	res := putCmdWithoutSigning{
+		apiObject: reportedAzureSEVSNPVersionAPI{Version: dateStr, AzureSEVSNPVersion: version},
+	}
+	return res.Execute(ctx, c.s3Client)
+}
+
+func (c Client) listReportedVersions(ctx context.Context) ([]string, error) {
+	list, err := c.s3Client.ListObjectsV2(ctx, &s3.ListObjectsV2Input{
+		Bucket: aws.String(c.bucketID),
+		Prefix: aws.String(reportVersionDir),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("list objects: %w", err)
+	}
+	var dates []string
+	for _, obj := range list.Contents {
+		fileName := path.Base(*obj.Key)
+		if strings.HasSuffix(fileName, ".json") {
+			dates = append(dates, fileName[:len(fileName)-5])
+		}
+	}
+	return dates, nil
+}
+
+// findMinVersion finds the minimal version of the given version dates among the latest values in the version window size.
+func (c Client) findMinVersion(ctx context.Context, versionDates []string) (*AzureSEVSNPVersion, string, error) {
```

Comment:
```suggestion
func (c Client) findMinVersion(ctx context.Context, versionDates []string) (AzureSEVSNPVersion, string, error) {
```

why make this a pointer?